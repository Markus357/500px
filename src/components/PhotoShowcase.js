import React from 'react';
import PropTypes from 'prop-types';
import Gallery from 'react-grid-gallery';
import ReactPaginate from 'react-paginate';
import styled from 'styled-components';

import WithImages from '../providers/WithImages';

const GalleryStyles = styled.div`
  .ReactGridGallery {
    margin: 2.5rem auto;
    text-align: center;
    box-shadow: 0px 0px 1.6rem 0px rgba(0,0,0,0.4);

    .tile {
      display: inline-block;
      float: none !important;
      outline: 3px solid #fff;
      box-shadow: 0px 0px 0px -4px rgba(0,0,0,0);
      transition: all ease-in 125ms;

      &:hover {
        z-index: 2;
        box-shadow: 0px 0px 30px -4px rgba(0,0,0,0.9);
        transform: scale( 1.15 );
      }
    }
  }
`;

const PaginationStyles = styled.div`
  position: relative;
  width: 100%;

  text-align: center;
  font-size: 1.6rem;
  line-height: 1.6rem;

  .pagination {
    display: inline-block;
    margin: 1rem auto;

    li {
      display: inline-block;

      text-align: center;

      margin: 0 0.4rem;
      border-radius: 0.3rem;

      &:hover { transform: scale( 1.2 ); }

      &.active {
        background-color: #000;

        a { color: #fff; }
      }

      &.previous, &.next, &.break { width: auto; }

      &.disabled {
        opacity: 0.4;

        a { cursor: not-allowed; }
      }

      a {
        cursor: pointer;
        display: inline-block;
        padding: 0.6rem;
        color: #000;
      }
    }
  }
`;

const Showcase = ({ images, totalPages, setImagePage }) => (
  <>
    { images && images.length &&
      <GalleryStyles>
        <Gallery
          enableImageSelection={ false }
          maxRows={ 3 }
          images={ images }
        />
      </GalleryStyles>
    }

    { totalPages > 1 &&
      <PaginationStyles>
        <ReactPaginate
          previousLabel={ 'previous' }
          nextLabel={ 'next' }
          breakLabel={ '...' }
          breakClassName={ 'break-me' }
          pageCount={ totalPages }
          marginPagesDisplayed={ 2 }
          pageRangeDisplayed={ 5 }
          onPageChange={ data => setImagePage({ pageNumber: data.selected +1 }) }
          containerClassName={ 'pagination' }
          subContainerClassName={ 'pages pagination' }
          activeClassName={ 'active' }
        />
      </PaginationStyles>
    }
  </>
);

Showcase.propTypes = {
  images: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      src: PropTypes.string.isRequired,
      thumbnail: PropTypes.string.isRequired,
      thumbnailWidth: PropTypes.number.isRequired,
      thumbnailHeight: PropTypes.number.isRequired,
      caption: PropTypes.string
    })
  ),
  totalPages: PropTypes.number,
  setImagePage: PropTypes.func,
};

// export both a hydrated and raw version of this component for easier testing
const PhotoShowcase = WithImages( Showcase );
const PhotoShowcaseRaw = Showcase;

export { PhotoShowcase, PhotoShowcaseRaw };
