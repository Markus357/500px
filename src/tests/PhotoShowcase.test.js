import React from 'react';
import renderer from 'react-test-renderer';

import images from './data/images-processed';
import { PhotoShowcaseRaw } from '../components/PhotoShowcase';

describe( '<PhotoShowcaseRaw />', () => {
  let defaultProps;
  let componentWithDefaultProps;
  let treeWithDefaultProps;

  beforeEach( () => {
    // reset defaults
    defaultProps = { images, totalPages: 10, currentPage: 1 };
    componentWithDefaultProps = renderer.create( <PhotoShowcaseRaw { ...defaultProps } /> );
    treeWithDefaultProps = componentWithDefaultProps.toJSON();
  });

  it( 'Should render the gallery when images are present', () => expect( treeWithDefaultProps ).toMatchSnapshot() );

  it( 'Should render pagination when there is more than 1 page of images', () => expect( treeWithDefaultProps ).toMatchSnapshot() );

  it( 'Should not render the gallery when images are empty', () => {
    // Modify default props for this scenario (empty images)
    const props = { ...defaultProps, images: [] };
    const component = renderer.create( <PhotoShowcaseRaw { ...props } /> );
    const tree = component.toJSON();

    expect( tree ).toMatchSnapshot();
  });
});