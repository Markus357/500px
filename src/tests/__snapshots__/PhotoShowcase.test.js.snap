// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhotoShowcaseRaw /> Should not render the gallery when images are empty 1`] = `
Array [
  "0",
  <div
    className="sc-bwzfXH gQlPco"
  >
    <ul
      className="pagination"
    >
      <li
        className="previous disabled"
      >
        <a
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          previous
        </a>
      </li>
      <li
        className="active"
      >
        <a
          aria-current="page"
          aria-label="Page 1 is your current page"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          1
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          aria-label="Page 2"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          2
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          aria-label="Page 3"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          3
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          aria-label="Page 4"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          4
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          aria-label="Page 5"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          5
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          aria-label="Page 6"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          6
        </a>
      </li>
      <li
        className="break-me"
      >
        <a
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          ...
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          aria-label="Page 9"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          9
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          aria-label="Page 10"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          10
        </a>
      </li>
      <li
        className="next"
      >
        <a
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          next
        </a>
      </li>
    </ul>
  </div>,
]
`;

exports[`<PhotoShowcaseRaw /> Should render pagination when there is more than 1 page of images 1`] = `
Array [
  <div
    className="sc-bdVaJa izVFFd"
  >
    <div
      className="ReactGridGallery"
      id="ReactGridGallery"
    >
      <iframe
        style={
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
            "height": 0,
            "margin": 0,
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "width": "100%",
          }
        }
      />
    </div>
  </div>,
  <div
    className="sc-bwzfXH gQlPco"
  >
    <ul
      className="pagination"
    >
      <li
        className="previous disabled"
      >
        <a
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          previous
        </a>
      </li>
      <li
        className="active"
      >
        <a
          aria-current="page"
          aria-label="Page 1 is your current page"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          1
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          aria-label="Page 2"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          2
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          aria-label="Page 3"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          3
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          aria-label="Page 4"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          4
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          aria-label="Page 5"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          5
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          aria-label="Page 6"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          6
        </a>
      </li>
      <li
        className="break-me"
      >
        <a
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          ...
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          aria-label="Page 9"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          9
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          aria-label="Page 10"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          10
        </a>
      </li>
      <li
        className="next"
      >
        <a
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          next
        </a>
      </li>
    </ul>
  </div>,
]
`;

exports[`<PhotoShowcaseRaw /> Should render the gallery when images are present 1`] = `
Array [
  <div
    className="sc-bdVaJa izVFFd"
  >
    <div
      className="ReactGridGallery"
      id="ReactGridGallery"
    >
      <iframe
        style={
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
            "height": 0,
            "margin": 0,
            "overflow": "hidden",
            "padding": 0,
            "position": "fixed",
            "width": "100%",
          }
        }
      />
    </div>
  </div>,
  <div
    className="sc-bwzfXH gQlPco"
  >
    <ul
      className="pagination"
    >
      <li
        className="previous disabled"
      >
        <a
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          previous
        </a>
      </li>
      <li
        className="active"
      >
        <a
          aria-current="page"
          aria-label="Page 1 is your current page"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          1
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          aria-label="Page 2"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          2
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          aria-label="Page 3"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          3
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          aria-label="Page 4"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          4
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          aria-label="Page 5"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          5
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          aria-label="Page 6"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          6
        </a>
      </li>
      <li
        className="break-me"
      >
        <a
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          ...
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          aria-label="Page 9"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          9
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          aria-label="Page 10"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          10
        </a>
      </li>
      <li
        className="next"
      >
        <a
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          next
        </a>
      </li>
    </ul>
  </div>,
]
`;
